reproject/mosaicking/coadd.py

```diff
@@ -154,10 +154,20 @@ def reproject_and_coadd(
         # each tile. Note that in future if we are worried about significant
         # distortions of the edges in the reprojection process we could simply
         # add arbitrary numbers of midpoints to this list.
-        ny, nx = array_in.shape
-        xc = np.array([-0.5, nx - 0.5, nx - 0.5, -0.5])
-        yc = np.array([-0.5, -0.5, ny - 0.5, ny - 0.5])
-        xc_out, yc_out = wcs_out.world_to_pixel(wcs_in.pixel_to_world(xc, yc))
+        if array_in.ndim == 2:
+            ny, nx = array_in.shape
+            xc = np.array([-0.5, nx - 0.5, nx - 0.5, -0.5])
+            yc = np.array([-0.5, -0.5, ny - 0.5, ny - 0.5])
+            xc_out, yc_out = wcs_out.world_to_pixel(wcs_in.pixel_to_world(xc, yc))
+            shape_out_cel = shape_out
+        elif array_in.ndim == 3:
+            nz, ny, nx = array_in.shape
+            xc = np.array([-0.5, nx - 0.5, nx - 0.5, -0.5])
+            yc = np.array([-0.5, -0.5, ny - 0.5, ny - 0.5])
+            zc = np.array([-0.5, nz - 0.5])
+            xc_out, yc_out = wcs_out.celestial.world_to_pixel(wcs_in.celestial.pixel_to_world(xc, yc))
+            zc_out = wcs_out.spectral.world_to_pixel(wcs_in.spectral.pixel_to_world(zc))
```

Comment:
Are the previous two lines done separately for memory efficiency?

Reply:
no, for readability - it's just easier for me to see this if they're separated into chunks.  Otherwise I think we need a full 2x3x3 grid with 0's appended on each side, which is going to be a lot of lines of code that don't do anything